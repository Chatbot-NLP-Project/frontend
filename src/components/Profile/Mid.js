import { useState, useEffect } from "react";
import Axios from "axios";

const Mid = () => {

    const [ChangeDetails, setChangeDetails] = useState(false);
    const [isPending, setIsPending] =  useState(false);

    const [email, setEmail] = useState('');
    const [firstName,setFirstName] = useState('');
    const [lastName,setLastName] = useState('');
    const [phoneNumber,setPhoneNumber] = useState('');

    useEffect(()=> {
        Axios.get("http://localhost:5000/profile", {
            params : {
                user_id:1   // set the user id by session_id
            }
          }).then((response) => {
              setFirstName(response.data.first_name);
              setLastName(response.data.last_name);
              setPhoneNumber(response.data.phone_number);
              setEmail(response.data.email);     
          }).catch((error) => {
              console.log(error);
          })
    },[ChangeDetails])

    const handleEditDetails = () => {
        setChangeDetails(true);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        setIsPending(true);
        Axios.post("http://localhost:5000/profile", {
                user_id:1,   // set the user id by session_id
                email: email,
                first_name: firstName,
                last_name: lastName,
                phone_number: phoneNumber,
          }).then((response) => {
            console.log(response);
            setChangeDetails(false);
        }).catch((error) => {
            console.log(error);
            setChangeDetails(false);
        })
        setIsPending(false);
        
    }

    return ( 
        <div className="details-content-mid">
            {(!ChangeDetails) ? (
                        <div className="details-content-right">
                        <div id="profile-details">
                        <h3>Profile Details</h3>
                            <button id="edit-details-btn" type="submit" onClick={handleEditDetails}>Edit Details</button>
                            <ul>
                            <li>First Name : <span>{firstName}</span></li>
                            <li>Last Name : <span>{lastName}</span></li>
                            <li>Phone Number : <span>{phoneNumber}</span></li>
                            <li>Email :  <span>{email}</span></li>
                            </ul>
                        </div>
                    </div>
            ) : (
                <div className="details-content-right">
                <div id="edit-details">
                    <h3>Edit Details</h3>
                    <form onSubmit={handleSubmit}>
                        <label>First Name</label>
                        <input 
                            type="text"
                            required
                            defaultValue = {firstName}
                            onChange = {(e) => setFirstName(e.target.value)}
                            // onChange = {(e) => console.log(e.target.value)}
                        />
                        <label>Last Name</label>
                        <input 
                            type="text"
                            required
                            value = {lastName}
                            onChange = {(e) => setLastName(e.target.value)}
                        />
                        <label>Phone Number</label>
                        <input 
                            type="text"
                            required
                            maxLength = {10}
                            minLength = {10}
                            value = {phoneNumber}
                            onChange = {(e) => setPhoneNumber(e.target.value)}
                        />
                        <label>Email</label>
                        <input 
                            type="text"
                            required
                            value = {email}
                            // onChange = {(e) => setEmail(e.target.value)}
                        />
                        {!isPending && <button>Save Details</button>}
                        {isPending && <button>Saving...</button>}
                    </form>
                </div>
            </div>
            )}
        </div>
 ); 
}
 
export default Mid;