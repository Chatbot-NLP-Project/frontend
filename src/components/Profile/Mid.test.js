// import dependencies
import React from 'react';
// import react dom
import ReactDOM from 'react-dom';
// import react-testing methods
import {render, cleanup, screen, fireEvent} from '@testing-library/react';
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from 'react-test-renderer';

import userEvent from '@testing-library/user-event';

// import the component for testing
import { MemoryRouter } from 'react-router-dom';
import Mid from "./Mid";

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MemoryRouter><Mid /></MemoryRouter>, div)
})

test('on initial render, edit details button is enabled', () => {
    const user = '{ "first_name":"Amara"}';
    localStorage.setItem('user', JSON.stringify(user));
    render(<Mid />)
    expect(screen.getByRole('button', {name: /edit details/i})).toBeEnabled();
})

test('immediately after clicking edit details, save details button is enabled', () => {
    const user = '{ "first_name":"Amara"}';
    localStorage.setItem('user', JSON.stringify(user));
    render(<Mid />)
    userEvent.click(screen.getByRole('button', {name: /edit details/i}));
    expect(screen.getByRole('button', {name: /save details/i})).toBeEnabled(); 
})

// it("renders footer correctly", () => {
//     const {getByTestId} = render(<MemoryRouter><Footer /></MemoryRouter>)
//     expect(getByTestId('heading-1')).toBeTruthy();
//     expect(getByTestId('heading-1')).toHaveTextContent("We will provide necessary informations that you want!!!");
//     // expect(getByTestId('subcription-text')).toBeTruthy();
//     // expect(getByTestId('subcription-text')).toHaveTextContent("Send us what you want to know");
//     // expect(getByTestId('send')).toBeTruthy();
//     // expect(getByTestId('send')).toHaveTextContent("Send us");
// })


it("matches snapshot", () => {
    const tree = renderer.create(<MemoryRouter><Mid /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})