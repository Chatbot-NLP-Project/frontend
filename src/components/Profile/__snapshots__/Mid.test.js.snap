// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="details-content-mid"
>
  <div
    className="details-content-right"
  >
    <div
      id="profile-details"
    >
      <h3>
        Profile Details
      </h3>
      <button
        id="edit-details-btn"
        onClick={[Function]}
        type="submit"
      >
        Edit Details
      </button>
      <ul>
        <li>
          First Name : 
          <span>
            
          </span>
        </li>
        <li>
          Last Name : 
          <span>
            
          </span>
        </li>
        <li>
          Phone Number : 
          <span>
            
          </span>
        </li>
        <li>
          Email :  
          <span>
            
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
